specification/sections/02-providers.md

```diff
@@ -225,9 +225,10 @@ class MyProvider implements Provider, AutoDisposable {
 
 #### Requirement 2.5.2
 
-> After a provider's shutdown function has terminated successfully, the provider's state **MUST** revert to its uninitialized state.
+> After a provider's shutdown function has terminated, the provider **SHOULD** revert to its uninitialized state.
```

Comment:
I've re-worded this to simply suggest that it's ideal (but not strictly required) that providers can be re-initialized after shutdown.

It's already possible some providers don't support this, and I think it's quite fine as long as they throw when re-initialized.